ci-operator/step-registry/aws/provision/tags-for-byo-vpc/aws-provision-tags-for-byo-vpc-commands.sh

```diff
@@ -13,24 +13,32 @@ fi
 
 REGION="${LEASED_RESOURCE}"
 
-infra_id=$(jq -r '.infraID' ${SHARED_DIR}/metadata.json)
 vpc_id=$(head -n 1 ${SHARED_DIR}/vpc_id)
 private_subnet_ids=$(yq-go r -j ${SHARED_DIR}/private_subnet_ids | jq -r '[ . | join(" ") ] | @csv' | sed "s/\"//g")
 public_subnet_ids=$(yq-go r -j ${SHARED_DIR}/public_subnet_ids | jq -r '[ . | join(" ") ] | @csv' | sed "s/\"//g")
-
-if [[ -z $vpc_id ]] || [[ -z $private_subnet_ids ]] || [[ -z $infra_id ]]; then
+if [[ -z $vpc_id ]] || [[ -z $private_subnet_ids ]]; then
   echo "Error: Can not get VPC id or private subnets, exit"
   echo "vpc: $vpc_id, private_subnet_ids: $private_subnet_ids"
   exit 1
 fi
 
-echo "Adding tags for VPC: $vpc_id, tags: kubernetes.io/cluster/${infra_id}, value: shared."
-aws --region $REGION ec2 create-tags --resources $vpc_id --tags Key=kubernetes.io/cluster/${infra_id},Value=shared
+# Add tags on VPC
+if [[ ${MANAGED_CLUSTER} == "no" ]]; then
+  infra_id=$(jq -r '.infraID' ${SHARED_DIR}/metadata.json)
+  if [[ -z $infra_id ]]; then
+    echo "Error: infra_id is empty"
+    exit 1
+  fi
+
+  echo "Adding tags for VPC: $vpc_id, tags: kubernetes.io/cluster/${infra_id}, value: shared."
+  aws --region $REGION ec2 create-tags --resources $vpc_id --tags Key=kubernetes.io/cluster/${infra_id},Value=shared
+fi
 
+# Add tags on subnets
 echo "Adding tags for private subnets:$private_subnet_ids, tags: kubernetes.io/role/internal-elb, value is empty."
 aws --region $REGION ec2 create-tags --resources $private_subnet_ids --tags Key=kubernetes.io/role/internal-elb,Value=
 
-if [[ ${APPLY_TAGS_TO_PUBLIC_SUBNETS} == "yes" ]]; then
+if [[ ${MANAGED_CLUSTER} == "yes" ]]; then
   echo "Adding tags for public subnets:$public_subnet_ids, tags: kubernetes.io/role/elb, value is empty."
   aws --region $REGION ec2 create-tags --resources $public_subnet_ids --tags Key=kubernetes.io/role/elb,Value=
 fi
```

Comment:
How about combine [L25-35](https://github.com/openshift/release/pull/41620/files#diff-d2363855530a9951bd709a70c003a445c28e71b7cf050ba815b21ee645288b77R25-R35) together:
```
if [[ ${MANAGED_CLUSTER} == "yes" ]]; then
  # Adding tag for public subnets
else
  # Adding infra_id tag for VPC
fi
```

Reply:
The reason why I do not combine these together is that they are two objects, one is vpc, another is subnet, there is no causation between the two. To be more scalability and readable, I'd like to put them into different blocks.

Reply:
Get it